api/submission/fetch_attempts_test.go

```diff
@@ -0,0 +1,75 @@
+package submission
+
+import (
+    "reflect"
+    "testing"
+
+    "github.com/eriq-augustine/autograder/api/core"
+    "github.com/eriq-augustine/autograder/model"
+    "github.com/eriq-augustine/autograder/util"
+)
+
+func TestFetchAttempts(test *testing.T) {
+    // Note that computation of these paths is deferred until test time.
+    studentGradingResults := []*model.GradingResult{
+        model.MustLoadGradingResult(getTestSubmissionResultPath("1697406256")),
+        model.MustLoadGradingResult(getTestSubmissionResultPath("1697406265")),
+        model.MustLoadGradingResult(getTestSubmissionResultPath("1697406272")),
+    };
+
+    testCases := []struct{
+        role model.UserRole
+        targetEmail string
+        foundUser bool
+        permError bool
+        result []*model.GradingResult
+    }{
+        // Grader, self.
+        {model.RoleGrader, "",                 true, false, []*model.GradingResult{}},
+        {model.RoleGrader, "grader@test.com",  true, false, []*model.GradingResult{}},
+
+        // Grader, other.
+        {model.RoleGrader, "student@test.com", true, false, studentGradingResults},
+
+        // Grader, missing.
+        {model.RoleGrader, "ZZZ@test.com",     false, false, []*model.GradingResult{}},
+
+        // Student, self.
+        {model.RoleStudent, "",                true, true, studentGradingResults},
```

Comment:
Make `result` nil in this case, since (unlike the other cases) this one will error out first.

Of course, the code you have works but this is a case of making things more clear to readers.